services/agents/mongodb.go

```diff
@@ -43,10 +50,22 @@ func mongodbExporterConfig(service *models.Service, exporter *models.Agent, reda
 		"--collect.topmetrics",
 		"--no-collect.connpoolstats",
 		"--no-collect.indexusage",
-
 		"--web.listen-address=:" + tdp.left + " .listen_port " + tdp.right,
 	}
 
+	env := []string{
+		fmt.Sprintf("MONGODB_URI=%s", exporter.DSN(service, time.Second, "")),
+		fmt.Sprintf("HTTP_AUTH=pmm:%s", exporter.AgentID),
+	}
+
+	// Starting with PMM 2.10.0, we are shipping the new mongodb_exporter
+	if !pmmAgentVersion.Less(newMongoExporterPMMVersion) {
+		args = []string{
+			"--compatible-mode",
+			"--web.listen-address=:" + tdp.left + " .listen_port " + tdp.right,
+		}
+	}
```

Comment:
I would change an order here.  Right now we have:
1. Flags for 2.9
2. Common environment variables.
3. Flags for 2.10.
4. Common flags.

That's a bit weird. How about moving environment variables before flags, and set flags in `if`? Like
```go
var args []string
if pmmAgentVersion.Less(newMongoExporterPMMVersion) {
  // flags for 2.9
} else {
  // flags for 2.10
}
```